Update backend to use COOL Terraform account

jsf9k opened the pull request:
## 🗣 Description

This pull request updates the backend configuration to use the COOL Terraform account instead of the playground account.  It also updates the README to agree with the Terraform code, and it renames two providers to give them more appropriate names.

## 💭 Motivation and Context

We should be using the COOL Terraform account for our backends, and the documentation should be current.

## 🧪 Testing

I checked that all the pre-commit hooks pass.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Update backend config to use the COOL Terraform account

jsf9k committed:
Rename two providers

The new names better reflect what the providers do.

jsf9k committed:
Improve description of one variable

jsf9k committed:
Update README.md to agree with Terraform code

felddy commented on backend.tf:
```diff
@@ -1,9 +1,9 @@
+    bucket         = "cisa-cool-terraform-state"
     dynamodb_table = "terraform-state-lock"
-    profile        = "playground"
+    role_arn       = "arn:aws:iam::608004238745:role/TerraformStateAccess"
```
Same question as before.  Profiles vs. ARNs in version control.

jsf9k replied:
> Same question as before. Profiles vs. ARNs in version control.

I had it pointing to profiles before, but I thought we decided to use ARNs instead.  See, for example, [this commit](https://github.com/cisagov/pca-terraform/pull/7/commits/2a0294214e5b921330c8b4afe7a85ea6968d64ea).

@dav3r may be able to shed some light on this.  His memory tends to be better than mine.

jsf9k replied:
After discussion with @felddy we decided to go back to using profiles.  See commits e5533b2 and e271f70.

jsf9k replied:
Using ARNs means one less layer of obfuscation, and one can look at the code in the repo and see what is going on without any knowledge of the way that we set up our boto3 profiles.

I think what makes the most sense is to use a profile to configure the backend (since variables are not allowed in that configuration block and we don't want actual ARN values out in the open) but use ARNs specified in variables (and hence in a `*.tfvars` file that is not committed) to configure the providers.  This may be what @felddy meant all along, and I simply misunderstood his intent.

See commit e5533b2.

felddy reviewed (approved):
Approved! See comment about profiles / arns.

jsf9k commented:
@felddy, I'm not entirely certain what to put for the `client_network` and `private_networks` inputs for [cisagov/openvpn-server-tf-module](https://github.com/cisagov/openvpn-server-tf-module).  I will just make them inputs and define them in my `variables.tf`, but I still need to know what values to put in my `*.tfvars` file at deployment time.

1. My thought is that `private_networks` should be list of networks actually assigned to subnets inside the VPC.  Is this correct?
2. My thought is that `client_network` should be some subnet of the VPC network that is not actually being used inside the VPC.  Is this correct, or can/should it be a subnet that does not overlap with the VPC network at all?

jsf9k committed:
Compute account number and subdomain from existing vars

We need the shared services account number as well as the public zone
name and OpenVPN server subdomain.  These values can be determined
from existing input variables.

jsf9k committed:
Add client_network and private_networks variables

jsf9k committed:
Use a profile instead of a role ARN to configure the backend

Putting account IDs and such in a public repo is not a big deal, but
some folks are already squeamish about using AWS so we may as well not
give them anything extra to worry about.

jsf9k committed:
Remove unnecessary file

mcdonnnj commented on README.md:
```diff
@@ -51,40 +44,30 @@ module "example" {
 | cert_bucket_name | The name of the AWS S3 bucket where certificates are stored. | string | `cool-certificates` | no |
-| cert_role_arn | The ARN of the role to assume when creating a role to allow reading certboto certificate data (e.g. "arn:aws:iam::123456789012:role/ReadCert-example.com"). | string | | yes |
+| cert_create_read_role_arn | The ARN of the role to assume when creating a role to allow reading certboto certificate data (e.g. "arn:aws:iam::123456789012:role/CertCreateReadRole"). | string | | yes |
+| client_network | A string containing the network and netmask to assign client addresses. The server will take the first address. (e.g. "10.240.0.0 255.255.255.0"). | string | | yes |
```
This input is marked as required but it is not in the example usage above.

jsf9k replied:
Good catch!  Resolved in commit 48850c3.

mcdonnnj commented on README.md:
```diff
@@ -51,40 +44,30 @@ module "example" {
-| master_subnet_id | The ID of the subnet where the IPA master is to be deployed (e.g. "subnet-0123456789abcdef0"). | string | | yes |
+| freeipa_admin_pw | The password for the FreeIPA Kerberos admin role. | string | | yes |
+| freeipa_client_security_group_id | The ID of the security group for FreeIPA clients. | string | | yes |
+| private_networks | A list of strings, each of which contains a network and netmask defining a list of subnets that exist behind the VPN server (e.g. ["10.224.0.0 255.240.0.0", "192.168.100.0 255.255.255.0"]).  These will be pushed to the clients. | list(string) | | yes |
```
This input is marked as required but it is not in the example usage above.

jsf9k replied:
Also resolved in commit 48850c3.

mcdonnnj reviewed (changes requested):
Looking good but please see my comments about the documentation.

jsf9k committed:
Add client_network and private_networks to example configuration

Thanks to @mcdonnnj for pointing that these required parameters were
missing!

dav3r reviewed (approved):
I'm late to the party here, but I don't see any issues at this point, other than waiting to hear back from @felddy about specifics for our Production `client_network` and `private_network` inputs.

mcdonnnj reviewed (approved):
Thanks for updating the documentation. Some strong work 💪
